compiler/rustc_ast_passes/messages.ftl

```diff
@@ -15,6 +10,16 @@ ast_passes_abi_custom_safe_function =
     functions with the `"custom"` ABI must be unsafe
     .suggestion = add the `unsafe` keyword to this definition
 
+ast_passes_abi_must_not_have_parameters_or_return_type=
+    invalid signature for `extern {$abi}` function
+    .note = functions with the `{$abi}` ABI cannot have any parameters or return type
```

Comment:
generally for style have been just formatting the ExternAbi (as it carries its quotation marks into formatting) without further ticks, or backticking around `extern {abi}`, I realize it's slightly inconsistent right now but all will be fixed in https://github.com/rust-lang/rust/pull/142134
```suggestion
    .note = functions with the {$abi} ABI cannot have any parameters or return type
```